pssh/ssh_client.py

```diff
@@ -335,3 +335,57 @@ def copy_file(self, local_file, remote_file, recurse=False):
         else:
             logger.info("Copied local file %s to remote destination %s:%s",
                         local_file, self.host, remote_file)
+
+    def _copy_dir_to_local(self, remote_dir, local_dir):
+        """Copies the remote directory to the local host."""
+        sftp = self._make_sftp()
+        file_list = sftp.listdir(remote_dir)
+        for file_name in file_list:
+            remote_path = os.path.join(remote_dir, file_name)
+            local_path = os.path.join(local_dir, file_name)
+            self.copy_file_to_local(remote_path, local_path, recurse=True)
+
+    def copy_file_to_local(self, remote_file, local_file, recurse=False):
+        """Copy remote file to local host via SFTP/SCP
+
+        Copy is done natively using SFTP/SCP version 2 protocol, no scp command \
+        is used or required.
+
+        :param remote_file: Remote filepath to copy the file from.
+        :type remote_file: str
+        :param local_file: Local filepath where the file will be copied.
+        :type local_file: str
+        :param recurse: Whether or not to recursively copy directories.
+        :type recurse: bool
+
+        :raises: :mod:'ValueError' when a directory is supplied to remote_file \
+        and recurse is not set
+        """
+        sftp = self._make_sftp()
+        try:
+            sftp.listdir(remote_file)
+            remote_dir_exists = True
+        except IOError or OSError:
+            remote_dir_exists = False
+        if remote_dir_exists and recurse:
+            return self._copy_dir_to_local(remote_file, local_file)
+        elif remote_dir_exists and not recurse:
+            raise ValueError("Recurse must be true if local_file is a "
+                             "directory.")
+        destination = [_dir for _dir in local_file.split(os.path.sep)
+                       if _dir][:-1][0]
+        if local_file.startswith(os.path.sep):
+            destination = os.path.sep + destination
+        if not os.path.exists(destination):
+            try:
+                os.makedirs(destination)
+            except OSError:
+                logger.error("Unable to create local directory structure.")
```

Comment:
Should fail here if parent directories cannot be created. Can just call `raise` to re-raise the `OSError`.

A test case of the 'cannot create parent directory structure' would be nice as well if I'm being picky - it would had caught this particular error :)